cheroot/connections.py

```diff
@@ -295,8 +294,17 @@ def _from_server_socket(self, server_socket):  # noqa: C901  # FIXME
 
             mf = MakeFile
             ssl_env = {}
+
             # if ssl cert and key are set, we try to be a secure HTTP server
             if self.server.ssl_adapter is not None:
+                # 1. Preemptive check for BuiltinSSLAdapter
+                if hasattr(
+                    self.server.ssl_adapter,
+                    '_check_for_plain_http',
+                ) and self.server.ssl_adapter._check_for_plain_http(s):
```

Comment:
Better move this into the wrap method of the adapter and raise a `NoSSLError` so that it's handled in the same `except` block below for both adapters.
This way, the adapter internals won't leak into this layer.